### PR TITLE
IGNITE-17212 Sql. Add support for DEFAULT operator

### PR DESCRIPTION
**Pull request by korlov42**

### CODE REVIEW

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/TypeUtils.java [hunk 1]
```diff
@@ -249,7 +251,9 @@ public static Object toInternal(ExecutionContext<?> ectx, Object val, Type stora
         } else if (storageType == LocalTime.class) {
             return (int) (((LocalTime) val).toNanoOfDay() / 1000 / 1000 /* convert to millis */);
```
Comment (AMashenkov) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/TypeUtils.java:252]:
```suggestion
            return (int) (TimeUnit.NANOSECONDS.toMillis(((LocalTime) val).toNanoOfDay()));
```

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/TypeUtils.java [hunk 2]
```diff
@@ -249,7 +251,9 @@ public static Object toInternal(ExecutionContext<?> ectx, Object val, Type stora
...
         } else if (storageType == LocalDateTime.class) {
-            return ((LocalDateTime) val).toEpochSecond(ZoneOffset.UTC);
+            var dt = (LocalDateTime) val;
+
+            return dt.toEpochSecond(ZoneOffset.UTC) * 1000 + dt.getNano() / 1000 / 1000 /* convert to millis */;
```
Comment (AMashenkov) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/TypeUtils.java:256]:
```suggestion
            return TimeUnit.SECONDS.toMillis(dt.toEpochSecond(ZoneOffset.UTC)) + TimeUnit.NANOSECONDS.toMillis(dt.getNano());
```

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/ddl/DdlSqlToCommandConverter.java [hunk 3]
```diff
@@ -557,4 +565,52 @@ private void checkPositiveNumber(int num) {
...
+     */
+    private static Object fromLiteral(RelDataType columnType, SqlLiteral literal) {
+        switch (columnType.getSqlTypeName()) {
+            case VARCHAR:
+            case CHAR:
+                return literal.getValueAs(String.class);
+            case DATE:
+                return LocalDate.ofEpochDay(literal.getValueAs(DateString.class).getDaysSinceEpoch());
+            case TIME:
+                return LocalTime.ofNanoOfDay(literal.getValueAs(TimeString.class).getMillisOfDay() * 1_000_000L);
```
Comment (AMashenkov) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/ddl/DdlSqlToCommandConverter.java:580]:
```suggestion
                return LocalTime.ofNanoOfDay(TimeUnit.MILLISECONDS.toNanos(literal.getValueAs(TimeString.class).getMillisOfDay()));
```

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java [hunk 4]
```diff
@@ -307,4 +312,131 @@ public void implicitPk() {
...
+            var createStatement = createStatementBuilder.append(")").toString();
+
+            sql(createStatement);
+            sql("INSERT INTO test (id) VALUES (0)");
+
+            var expectedVals = args.stream()
+                    .map(a -> a.expectedVal)
+                    .collect(Collectors.toList());
+
+            assert args.size() > 0;
```
[the remark this answers is not in the document]
Comment (korlov42) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java]: done

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/TypeUtils.java (hunk 1 above)
Comment (korlov42) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/TypeUtils.java]: fixed

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/TypeUtils.java (hunk 2 above)
Comment (korlov42) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/TypeUtils.java]: fixed

**Pull request merged**